lib/active_publisher/async/in_memory_adapter/async_queue.rb

```diff
@@ -0,0 +1,63 @@
+module ActivePublisher
+  module Async
+    module InMemoryAdapter
+      class AsyncQueue
```

Comment:
My hesitation with calling it a `Queue` is that it's not _just_ a `Queue`. It uses a `Queue`, though. I think `SupervisedPublisherQueue` would be a more appropriate name. But I'm fine with leaving it the way it is.